sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -121,8 +123,13 @@ class SessionCatalog(
     currentDb = db
   }
 
-  def getDefaultDBPath(db: String): String = {
-    System.getProperty("java.io.tmpdir") + File.separator + db + ".db"
+  def getDatabasePath(dbName: String, path: Option[String]): String = {
+    val dbPath =
+      path.map(new Path(_))
+        .getOrElse(
+          new Path(new Path(System.getProperty("java.io.tmpdir")), dbName.toLowerCase() + ".db"))
+    val dbLocation = dbPath.toString
+    if (dbLocation.endsWith(File.separator)) dbLocation.dropRight(1) else dbLocation
```

Comment:
Actually, `getDatabasePath` in `SessionCatalog.scala` is not used. We are not creating any physical directory for this. Thus, I assume here the directory path is based on the OS platform. Personally, I think this should return an empty string.

In `HiveSessionCatalog`, what we did is based on Hadoop FS interface.